Sources/SwiftBSON/BSONDocumentIterator.swift

```diff
@@ -22,12 +22,10 @@ public class BSONDocumentIterator: IteratorProtocol {
 
```

Comment:
A lot of the changes in this file are for making it safe to use an iterator with invalid BSON, defaulting to just returning early or nil when errors are encountered.